mds/apis/agency_api/v0_x/compliance.py

```diff
@@ -0,0 +1,119 @@
+from rest_framework import serializers
+from rest_framework import viewsets
+
+from django.db.models import Q, Prefetch
+
+from mds import models
+
+from datetime import datetime
+import pytz
+
+
+class ComplianceSerializer(serializers.ModelSerializer):
+    rules_id = serializers.SerializerMethodField()
+    compliances = serializers.SerializerMethodField()
+
+    class Meta:
+        model = models.Policy
+        fields = ("id", "rules_id", "compliances")
+
+    def get_rules_id(self, policy):
+        return [rule["rule_id"] for rule in policy.rules]
+
+    def get_compliances(self, policy):
+        final_compliance = []
+        for compliance in policy.compliances_pref:
+            for current_compliance in final_compliance:
+                if current_compliance["rule_id"] == compliance.rule:
+                    for match in current_compliance["matches"]:
+                        if match["geography"] == str(compliance.geography):
+                            match["measured"] += 1
+                            break
+                    else:
+                        current_compliance["matches"].append(
+                            {"geography": str(compliance.geography), "measured": 1}
+                        )
+                    current_compliance["vehicles_in_violation"].append(
+                        str(compliance.vehicle.id)
+                    )
+                    current_compliance["total_violations"] = len(
```

Comment:
You may compute every "total_violations" at the end of your function
This would avoid getting through every vehicles in violations many many times